File: 0366-Find Leaves of Binary Tree/0366-Find Leaves of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
 class Solution {
public:
    vector<vector<int>> findLeaves(TreeNode* root) {
        vector<vector<int>> result;
        height(root, result);
        return result;
    }
    
private:
    int height(TreeNode* root, vector<vector<int>>& result) {
        if (root == nullptr) {
            return -1;
        }
        
        int leftHeight = height(root->left, result);
        int rightHeight = height(root->right, result);
        int h = 1 + max(leftHeight, rightHeight);
        if (result.size() <= h) {
            result.push_back({});
        }
        result[h].push_back(root->val);
        return h;
    }
};

File: 0301-0400/0366-Find Leaves of Binary Tree/0366-Find Leaves of Binary Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def findLeaves(self, root: TreeNode) -> List[List[int]]:
        result = []
        def height(root):
            if root is None:
                return -1
            leftHeight = height(root.left)
            rightHeight = height(root.right)
            h = 1 + max(leftHeight, rightHeight)
            if len(result) <= h:
                result.append([])
            result[h].append(root.val)
            return h
        height(root)
        return result
